src/osd/ReplicatedPG.cc

```diff
@@ -5209,11 +5212,9 @@ int ReplicatedPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	}
 
 	t->truncate(soid, op.extent.offset);
-	if (oi.size > op.extent.offset) {
-	  interval_set<uint64_t> trim;
-	  trim.insert(op.extent.offset, oi.size-op.extent.offset);
-	  ctx->modified_ranges.union_of(trim);
-	}
+	//need recover whole object after truncate Ops
+	ctx->modified_ranges.clear();
+	ctx->modified_ranges.insert(0, (uint64_t) - 1);
```

Comment:
it does not make sense except for memory consumption. If this will be a issue, maybe uint32_t will be a better option?